src/language/isolates.md

```diff
@@ -0,0 +1,802 @@
+    final jsonData = jsonDecode(fileData) as Map<String, dynamic>;
+    return jsonData;
+  });
+
+  // Use that data.
+  print('Number of JSON keys: ${jsonData.length}');
+}
+```
+
+This example accomplishes the same as the previous.
+A new isolate spawns, computes something, and sends back the result.
+
+However, now the isolate sends a [closure][].
+Closures are less limited than typical named functions,
+both in how they function and how they're written into the code.
+In this example, `Isolate.run()` executes what looks like local code, concurrently.
+In that sense, you can imagine `run()` to work like a control flow operator
+for "run in parallel".
+
+[closure]: /language/functions#anonymous-functions
+
+## Sending multiple messages between isolates with ports
+
+`Isolate.run()` abstracts a handful of lower-level,
+isolate-related API to simplify isolate management:
+
+* [`Isolate.spawn()`][] and [`Isolate.exit()`][]
+* [`ReceivePort`][] and [`SendPort`][]
+* [`SendPort.send()` method][]
+
+[`Isolate.exit()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/exit.html
+[`Isolate.spawn()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/spawn.html
+[`ReceivePort`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/ReceivePort-class.html
+[`SendPort`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/SendPort-class.html
+[`SendPort.send()` method]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}
+/dart-isolate/SendPort/send.html
+
+### ReceivePorts and SendPorts
+
+This long-lived communication between isolates is set up with two classes (in addition to Isolate): ReceivePort and SendPort. These ports are the only way isolates can communicate with each other.
+
+Ports behave similarly to Streams, in which the StreamController or Sink is 
+created in one isolate, and the listener is set up in the other isolate. In 
+this analogy, the StreamController is called a SendPort, and you can “add” 
+messages with the [`send()` method][]. ReceivePorts are the listeners, and 
+when these listeners receive a new message, they call a provided callback with the message as an argument.
+
+[`send()` method]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}
+/dart-isolate/SendPort/send.html
+
+#### Setting up ports
+
+Setting up a port requires a few steps, which must be done in a specific order. The following figure shows the conceptual steps required to spawn a new isolate and establish 2-way communication between it and the main isolate.
+
+{{site.alert.note}}
+This diagram, and the following diagram, are high-level and intended to convey the concepts necessary to use isolates, but actual implementation requires a bit more code. A full code example is shown later on this page.  
+{{site.alert.end}}
+
+![A figure showing events being fed, one by one, into the event loop](/assets/img/language/concurrency/ports-setup.png)
```

Comment:
Hate to do this because the diagrams look great, but could you take the numbered list out of the diagram and just have that as text below? I think it's an accessibility thing (it's not screen readable) and also not searchable (this one and line 200, sorry!)